### PR TITLE
Appium set device properties

### PR DESCRIPTION
**Pull request by ncalaway**
Get and set networking properties on a mobile device

### CODE REVIEW

**Comment by ncalaway** [comment]
This would probably be just for Appium. I know Puppeteer can test mobile devices in the sense that Chrome DevTools can render different sizes for different devices. However, the devices in DevTools are really just different sized viewports, without anything else to them. Correct me if I'm wrong.

File: src/appium/appium-helpers.ts [hunk 1]
```diff
@@ -91,6 +91,11 @@ export const appiumSessionCreate = (scenario: Scenario, opts: any = {}) => {
   return async () => {
     scenario.set("capabilities", opts);
     const newSessionId = await createAppiumSession(scenario, opts);
+    const capabilities = await getAppiumSessionCapabilities(
+      newSessionId,
+      scenario
+    );
+    scenario.set("capabilities", capabilities);
```
Comment (jasonbyrne) [inline src/appium/appium-helpers.ts:98]: I'm really not a fan of setting these session-level variables

**Review by jasonbyrne — CHANGES_REQUESTED** [review]
I would like the scenario.set stuff to be removed

**Review by jasonbyrne — APPROVED** [review]

File: src/appium/appium-helpers.ts [hunk 2]
```diff
@@ -213,13 +218,142 @@ const getAppiumSessionCapabilities = async (
...
-const setDevProperties = async (
+export const setDevProperties = async (
   sessionId: string,
-  scenario: Scenario,
+  scenario: iScenario,
   devProperties: DeviceProperties = {}
-): Promise<any> => {
+): Promise<void> => {
+  console.log("setDevProperties");
+  console.log(devProperties);
```
Comment (jasonbyrne) [inline src/appium/appium-helpers.ts:229]: Should remove these console logs

File: src/appium/appium-helpers.ts (hunk 1 above)
Comment (jasonbyrne) [reply src/appium/appium-helpers.ts:98]: Shouldn't these `scenario.set`s be removed?

File: src/appium/appiumresponse.ts [hunk 3]
```diff
@@ -396,4 +498,8 @@ export class AppiumResponse extends ProtoResponse implements iResponse {
       }
     );
   }
+
+  public destroySession() {
+    console.log("hello");
```
Comment (jasonbyrne) [inline src/appium/appiumresponse.ts:503]: What's this for?

**Review by jasonbyrne — CHANGES_REQUESTED** [review]

**Review by jasonbyrne — APPROVED** [review]

**Pull request merged**